durabletask/worker.py

```diff
@@ -517,12 +564,32 @@ def process_event(self, ctx: _RuntimeOrchestrationContext, event: pb.HistoryEven
                             f"{ctx.instance_id}: Ignoring unexpected timerFired event with ID = {timer_id}.")
                     return
                 timer_task.complete(None)
-                ctx.resume()
+                if timer_task._retryable_parent is not None and timer_task._retryable_type is not None:
+                    activity_action = timer_task._retryable_parent._action
+                    
+                    if timer_task._retryable_type == "activity":
```

Comment:
I'm wondering if there's a way to simplify the design so that we don't need the `_retryable_type` field. It feels a bit fragile. I don't have any specific suggestions right now, but I'd like us to consider it.

Reply:
Have tried to modify the logic, to look more standarized. Using an enum and setting it as a part of RetryableTask.

Reply:
Modified to use a boolean, instead of enum.